### PR TITLE
bootstrap the root workspace only on the root shard

### PR DESCRIPTION
**Pull request by p0lyn0mial**
## Summary
the root workspace is only available on the root shard, this pr makes sure it will be only created on it.

## Related issue(s)

Fixes #

### CODE REVIEW

File: pkg/server/server.go [hunk 1]
```diff
@@ -417,18 +417,20 @@ func (s *Server) Run(ctx context.Context) error {
...
 
-		// bootstrap root workspace phase 0, no APIBinding resources yet
-		if err := configrootphase0.Bootstrap(goContext(ctx),
-			kcpClusterClient.Cluster(tenancyv1alpha1.RootCluster),
-			apiextensionsClusterClient.Cluster(tenancyv1alpha1.RootCluster).Discovery(),
-			dynamicClusterClient.Cluster(tenancyv1alpha1.RootCluster),
-		); err != nil {
-			// nolint:nilerr
-			return nil // don't klog.Fatal. This only happens when context is cancelled.
+		if s.options.Extra.ShardName == tenancyv1alpha1.RootCluster.String() {
```
Comment (sttts) [inline pkg/server/server.go:420]: nit: should we introduce a RootShard constant? This is not really the cluster.

Comment (p0lyn0mial) [reply pkg/server/server.go:420]: why not

Comment (p0lyn0mial) [reply pkg/server/server.go]: done.

**Pull request merged**